components/service/glean/src/main/java/mozilla/components/service/glean/debug/GleanDebugActivity.kt

```diff
@@ -33,16 +43,37 @@ class GleanDebugActivity : Activity() {
 
         // Enable debugging options and start the application.
         intent.extras?.let {
+            // Check for ping debug view tag to apply to the X-Debug-ID header when uploading the
+            // ping to the endpoint
+            var pingTag: String? = intent.getStringExtra(TAG_DEBUG_VIEW_EXTRA_KEY)
+            var altUrl: String? = null
+
+            // Validate the ping tag against the regex pattern
+            pingTag?.let {
+                if (!pingTagPattern.matches(it)) {
+                    logger.error("tagPings value $it does not match accepted pattern $pingTagPattern")
+                    pingTag = null
```

Comment:
We should also reset the `altUrl` to the previously set value. Alternatively, we could change how things are handled in `HttpPingUploader.buildRequest`: if there's a tag, use this endpoint, otherwise, use the one in configuration.

Reply:
Unfortunately, if I try and intercept the endpoint in the HttpPingUploader, it's too late to redirect it for tests and I have no way to capture a ping in a test to evaluate the contents.  I tried this originally but realized it made very little difference (functionally) where it got changed and this allowed me to still change the endpoint before WorkManager is triggered.

Reply:
```kotlin
serverEndpoint = altUrl?.let { it } ?: Glean.configuration.serverEndpoint
```
This line should make it so we don't have to reset altUrl to the previous.   If `pingTag` isn't there or if it doesn't match the regex, then altUrl will be null and it will just use the current configuration endpoint.

Reply:
Yes, this is true the first time this is called, but we can't really control what users will use this. Think of this sequence of actions:

- App is started (`config.serverEndpoint = <original server>`)
- Tagging with `X-Debug-ID=test` (`config.serverEndpoint = <tagging server>`)
- Tagging with `X-Debug-ID=£$"(/£$)"£($". The value is not valid ,so `pingTag` gets set to `null`. However `altUrl` is not reset (`config.serverEndpoint = <tagging server>`)

We are now in a mixed condition: if the app continues to be used, even if no tag is applied, data will be sent to the tagging endpoint.

Reply:
@Dexterp37 Since the ONLY time that the intent extras are read is in onCreate(), and since altUrl is a local variable to that function, I don't think I need to do anything here.  The ONLY way that they could be re-tagged is by recreating the debug activity, which would start up with altUrl = null, and then just use the default endpoint from the configuration.

Basically, it doesn't look like it's possible to send a new tag without tearing down the debug activity and recreating it.

Reply:
Per our convo in slack I think I have this addressed now